### PR TITLE
x11-terms/guake: add 3.9.0, enable py3.11

### PR DESCRIPTION
**Pull request by corvus1**
Simple version bump, new dep added https://github.com/Guake/guake/releases/tag/3.9.0

UPD: tested and enabled py3.11. Buy one, get one for free. :-P

Closes: https://bugs.gentoo.org/889900
Closes: https://bugs.gentoo.org/890259

Signed-off-by: Michael Corvinus <voron1@gmail.com>

### CODE REVIEW

**Comment by scardracs** [comment]
@mattst88 could be a good idea to add gnome project as maintainer?

File: x11-terms/guake/guake-3.9.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_SINGLE_IMPL=1
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{8..10} )
```
Comment (scardracs) [inline x11-terms/guake/guake-3.9.0.ebuild:8]: Doesn't it works against python 3.11?

File: x11-terms/guake/guake-3.9.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+PLOCALES="ca cs de el es fa fi fr gl hr hu id it ja ko nb nl pa pl pt_BR ru sv tr uk zh_CN zh_TW"
+
+DESCRIPTION="Drop-down terminal for GNOME"
+HOMEPAGE="http://guake-project.org/"
+SRC_URI="https://github.com/Guake/guake/archive/refs/tags/${PV}.tar.gz -> ${P}.gh.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="amd64 arm ~arm64 ~ppc64 x86"
```
Comment (scardracs) [inline x11-terms/guake/guake-3.9.0.ebuild:19]: These should be all unstable

File: x11-terms/guake/guake-3.9.0.ebuild (hunk 1 above)
Comment (corvus1) [reply x11-terms/guake/guake-3.9.0.ebuild:8]: > Doesn't it works against python 3.11?

Untested. I am on 3.10 currently. 
Reluctant to switch. Please don't ask me? :-)

File: x11-terms/guake/guake-3.9.0.ebuild (hunk 2 above)
Comment (corvus1) [reply x11-terms/guake/guake-3.9.0.ebuild]: > These should be all unstable

Fixed.

File: x11-terms/guake/guake-3.9.0.ebuild (hunk 1 above)
Comment (scardracs) [reply x11-terms/guake/guake-3.9.0.ebuild:8]: Python 3.11 will be the stable version in no time: you'd better test it

Comment (scardracs) [reply x11-terms/guake/guake-3.9.0.ebuild:8]: I have no problem against it. Mine was a suggestion just to drop the future PR but you may want to ask to a dev too.

Comment (corvus1) [reply x11-terms/guake/guake-3.9.0.ebuild:8]: The package is currently utterly abandoned and unmaintained. I am using it on my laptop, but my laptop was also abandoned and broken for a while. I recently repaired it enough for it to start working, and now I'm trying to bring it back into some kind of decent condition. New ssd, new fan, new battery... new version of guake. If both survive the process, they will live. :-) The laptop still needs a new wifi adapter and a new rechargeable RTC battery. And guake needs support for new Python. The surgery is still in progress. :-)

Comment (corvus1) [reply x11-terms/guake/guake-3.9.0.ebuild:8]: > but you may want to ask to a dev too.

Did you mean that I could join the developers and become a developer myself?

Comment (scardracs) [reply x11-terms/guake/guake-3.9.0.ebuild:8]: > > but you may want to ask to a dev too.
> 
> Did you mean that I could join the developers and become a developer myself?

No. I mean you have to ask to a dev if it is ok to add python3.11 in a 2nd PR

Comment (corvus1) [reply x11-terms/guake/guake-3.9.0.ebuild:8]: Ah, okay. I got confused for a second there. :-)

**Comment by corvus1** [comment]
@scardracs I tested guake with py3.11 and it works.
Can I now get my LGTMs? :-) On this one and on double-commander, ideally. That would be so nice.

**Comment by scardracs** [comment]
> @scardracs I tested guake with py3.11 and it works.
> 
> Can I now get my LGTMs? :-) On this one and on double-commander, ideally. That would be so nice. 

It LGTM. Now you have to wait for a dev

**Comment by scardracs** [comment]
Be sure to add Closes: https://bugs.gentoo.org/890259 on your PR ;)

**Comment by corvus1** [comment]
> Be sure to add
> 
> ```
> Closes: https://bugs.gentoo.org/890259 
> Closes: https://bugs.gentoo.org/890259
> ```
> 
> on your PR ;)

Yeah, thanks for linking it, but it is my bug, and it is linked both ways, as "see also" and "pull request" keyword in the bug, and "Closes:" in the commit message. I was very particular about doing that.

**Comment by scardracs** [comment]
> ## Pull Request assignment
> 
> _Submitter_: @corvus1 _Areas affected_: ebuilds _Packages affected_: x11-terms/guake
> 
> **x11-terms/guake**: @gentoo/proxy-maint (maintainer needed)
> ## Linked bugs
> 
> Bugs linked: [890259](https://bugs.gentoo.org/890259), [889900](https://bugs.gentoo.org/889900)
> 
> In order to force reassignment and/or bug reference scan, please append `[please reassign]` to the pull request title.
> 
> _Docs_: [Code of Conduct](https://wiki.gentoo.org/wiki/Project:Council/Code_of_conduct) ● [Copyright policy](https://www.gentoo.org/glep/glep-0076.html) ([expl.](https://dev.gentoo.org/~mgorny/articles/new-gentoo-copyright-policy-explained.html)) ● [Devmanual](https://devmanual.gentoo.org/) ● [GitHub PRs](https://wiki.gentoo.org/wiki/Project:GitHub/Pull_requests) ● [Proxy-maint guide](https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide)

Oh, I hadn't noticed this one ':)

**Comment by corvus1** [comment]
It's just a little demotivating to me, that I get lgtms, and then it hangs for weeks. I feel like I'm being ignored by the community, while actively contributing volunteer work to it.

**Comment by scardracs** [comment]
> It's just a little demotivating to me, that I get lgtms, and then it hangs for weeks. I feel like I'm being ignored by the community, while actively contributing volunteer work to it.

The point is that devs that can merge all these PRs can be count on one hand. I used to have some PRs opened for months but it's not your or their fault

**Comment by corvus1** [comment]
Yeah, I get all that. But I heard that python team is all over switching to 3.11 by default, so enabling py3.11 should be a pretty big priority. And all changes here are trivial, and already reviewed. It's trivial. It'd take any dev seconds, a minute tops. Right?

And people have been known to do drive-by things like that. Ionen bumped guake a couple of times, but the last time I think he wrote in the commit message that he's not going to anymore. Sam loves doing drive-by commits, fixing random things. 

I don't know, I'm just ranting now. I should probably stop. :-)

File: x11-terms/guake/guake-3.9.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+PLOCALES="ca cs de el es fa fi fr gl hr hu id it ja ko nb nl pa pl pt_BR ru sv tr uk zh_CN zh_TW"
+
+DESCRIPTION="Drop-down terminal for GNOME"
+HOMEPAGE="http://guake-project.org/"
+SRC_URI="https://github.com/Guake/guake/archive/refs/tags/${PV}.tar.gz -> ${P}.gh.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+IUSE="utempter"
```
Comment (juippis) [inline x11-terms/guake/guake-3.9.0.ebuild:20]: This use flag has no function in this ebuild, we need to move it under `optfeature` instead.
https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

I can do it.

**Review by juippis — APPROVED** [review]

**Comment by juippis** [comment]
Reviewing PRs always takes time. And we have to build-test it, this pulled ~80 deps so it takes a while. 

Sorry for the wait, especially during holiday seasons the backlog grows so it takes a while to catch it up again. And we, or at least I, try to process these PRs in order. First created first merged. During "normal times" PRs are usually processed within 3 days.

As to pushing new packages via proxy-maint project, they're most likely ignored. Please see 
https://archives.gentoo.org/gentoo-proxy-maint/message/44f7712fb49850288cd840c3541f6d7e
https://archives.gentoo.org/gentoo-proxy-maint/message/e5556fbdbd26b05636be478853ff44b2
for some history regarding that. If your new package isn't a "popular" one you'll most likely need to find an individual dev to proxy it for you, since the proxy-maint project has hands full dealing with existing packages already.

With all that said, thanks for bumping guake!

**Comment by corvus1** [comment]
> With all that said, thanks for bumping guake!

Sure! I'm happy to help. I love this little thing. It has "maintainer wanted" written all over it, for years, but somehow I think it's pretty popular. It's strange. Another example of a popular python package is Thonny https://thonny.org/ https://bugs.gentoo.org/666802 I'd be glad to add it, and proxied-maintain it, but it seems really hard to do. Like... I think I have the tools to be dangerous and do all the work, but then I'll just get stuck. I'm not sure how to proceed here.

**Comment by juippis** [comment]
Could you get more familiar with the guru project? You could first try adding the ebuild to the guru repository, and then we'd see how it's maintained there etc if people like it or not.

**Comment by corvus1** [comment]
>  If your new package isn't a "popular" one you'll most likely need to find an individual dev to proxy it for you, since the proxy-maint project has hands full dealing with existing packages already.

As to the less popular package that I really wanted to add in my other PR, I sorta had a talk with Sam, and he said he's "not opposed" to it. But, I guess there's only so much that can be done. We're always in the middle of something. The holidays, the great breakage of llvm-16, then there's going to be something else.

**Comment by corvus1** [comment]
> Could you get more familiar with the guru project? You could first try adding the ebuild to the guru repository, and then we'd see how it's maintained there etc if people like it or not.

Does that ever actually happen? Importing new packages from ::guru?

**Comment by juippis** [comment]
Yes, but honestly we are taking the cherries on top. There's too much offering, realistically not everything can be imported.

**Comment by corvus1** [comment]
So, here's the thing I perhaps don't fully understand. We get cited the lack of manpower, which is understandable. But we are contributing exactly that. Work. I'll have to assume that there is some kind of different work that has to be done so gentoo can utilize ours. The question is, can we somehow help with that?

**Comment by juippis** [comment]
**If** a contribution is of good quality, adhering to all our policies, no build failures with any use flag combinations etc it's a fast merge from our side. But often they're not. And we have to "train" every new contributor pretty much, then they do 1-2 contributions and are never seen again. Then pops a new guy that we have to train again. 

Even if you look at this PR there were some small things that weren't perfect, which were caught. And even when you learn the devmanual, PMS, our policy guides so **your** contributions are always top-notch, there may still be 20 PRs in front of yours that get reviewed first. I personally try to process these in the order they were opened, to be fair, and sometimes reviewing 1 hard PR takes more time than reviewing 10 easy ones. 

If you want to help, you'll learn devmanual, PMS, all commit-related GLEPs and policies, and go through the hard PRs before devs get to there ;) but gaining the required knowledge will take a lot of time, since Gentoo as a project is so vast.
(of course you can focus on areas closer to your personal interests and strengths)

**Comment by corvus1** [comment]
> **If** a contribution is of good quality, adhering to all our policies, no build failures with any use flag combinations etc it's a fast merge from our side. But often they're not. 

That is easy to understand. And I was trying. I took the ebuild, and I renamed it (how hard can I mess that up, right?).
Then I realized, that I need to add a new dep. Managed to actually mess that up a little, but very quickly corrected myself. Now all the deps look the same. So, basically, the in-tree ebuild is my baseline of quality. If that is accepted in-tree, as a baby-maintainer, I emulate. That's what babies do. 

Then turns out the `utemper` thing was wrong to begin with. That I of course couldn't know, without extensive knowledge of devmanual and policies. 

> EDIT: Right, we do have to build-test any contribution to make sure nothing obvious bad happens. If there's giant packages like qtwebengine etc being pulled... .... ... yeah.

Oh, let me assure you, that nothing gets pull-requested without being built and ran from my ::local. I build it, and make sure it has at least some signs of life in runtime. That's the simpliest test one can always do.

> 
> Even if you look at this PR there were some small things that weren't perfect, which were caught. And even when you learn the devmanual, PMS, our policy guides so **your** contributions are always top-notch, there may still be 20 PRs in front of yours that get reviewed first. I personally try to process these in the order they were opened, to be fair, and sometimes reviewing 1 hard PR takes more time than reviewing 10 easy ones.
> 
> If you want to help, you'll learn devmanual, PMS, all commit-related GLEPs and policies, and go through the hard PRs before devs get to there ;) but gaining the required knowledge will take a lot of time, since Gentoo as a project is so vast. (of course you can focus on areas closer to your personal interests and strengths)

So basically the recommendation is to become better. That doesn't scare me. I like getting better. There's still the question of evaluation. So I can figure out if I'm getting better. Like, I get LGTMs. Is it the same standard, that will be accepted into the tree, or is there still some developer's work?

**Comment by juippis** [comment]
> So basically the recommendation is to become better. That doesn't scare me. I like getting better. There's still the question of evaluation. So I can figure out if I'm getting better. Like, I get LGTMs. Is it the same standard, that will be accepted into the tree, or is there still some developer's work?

If you become an active contributor, and never/rarely need guidance, we do tend to trust you more and review things less strictly. But I'd also advise to stick things close to your own interests, i.e. packages that you **use** and lack maintenance. Also not everyone uses Github, so Github contributions can be rather 50-50 whether the maintainer gets notified about them or not. If you don't receive an answer from the maintainer, I recommend opening a bug and linking your PR commit to a bug (with `Bug:` or `Closes:` tag like you've done here) to get maintainer's attention. proxy-maint and maintainer-needed packages are reviewed here, so generally you should begin "fixing" those packages until you learn better which teams respond here and which do not. 

If you've opened a bug and provided a contribution via Github that receives no attention (for say 4 weeks), please join #gentoo-dev-help IRC channel and ping someone to take a look. Do note that packages assigned to base-system or toolchain **can not** be touched by members outside those projects, so for those, no one can help you. But those projects are not great for first-time-contributions anyway ;)

**Comment by corvus1** [comment]
> But those projects are not great for first-time-contributions anyway ;)

I think for my first contribution I actually did pretty okay. :-) I have a feeling Andreas had something to do with it, even though he didn't say anything. :-D
